src/pretix/helpers/hierarkey.py

```diff
@@ -0,0 +1,17 @@
+def clean_filename(fname):
+    """
+    hierarkey.forms.SettingsForm appends a random value to every filename. However, it keeps the
+    extension around "twice". Tis leads to:
+
+    "Terms.pdf" → "Terms.pdf.OybgvyAH.pdf"
+
+    In pretix Hosted, our storage layer also adds a hash of the file to the filename, so we have
+
+    "Terms.pdf" → "Terms.pdf.OybgvyAH.22c0583727d5bc.pdf"
+
+    This function reverses this operation:
+
+    "Terms.pdf.OybgvyAH.22c0583727d5bc.pdf" → "Terms.pdf"
+    """
+    ext = fname.split('.')[-1]
+    return fname.rsplit(ext, 2)[0] + ext
```

Comment:
What happens if the original file did not have a filename-extension? Then the new extension would be `22c0583727d5bc`. Not really a problem, but just to make sure. 

Also I think you mean to only split once (so you get 2 parts), not twice (which could lead to 3 parts). And also make sure to split at an „ext-point“ separated by a "." and lateron add the ext separated by a ".":

```suggestion
    ext = '.' + fname.split('.')[-1]
    return fname.rsplit(ext, 1)[0] + ext
```

Reply:
> What happens if the original file did not have a filename-extension? 

Then we would never allow for that file to be uploaded in the first place.  But yes, we can apply your fix.

> Also I think you mean to only split once (so you get 2 parts), not twice (which could lead to 3 parts). And also make sure to split at an „ext-point“ separated by a "." and lateron add the ext separated by a ".":

Nope. Pretty sure my code works correctly. With my code,
``clean_filename("fooasdas.asdasd.pdf.0123456.abcdef00.pdf") == "fooasdas.asdasd.pdf"``.

With your code, ``clean_filename("fooasdas.asdasd.pdf.0123456.abcdef00.pdf") == "fooasdas.asdasd.pdf.0123456.abcdef00.pdf"``.

This is because of:

```
>>> "fooasdas.asdasd.pdf.0123456.abcdef00.pdf".rsplit(".pdf", 1)
['fooasdas.asdasd.pdf.0123456.abcdef00', '']
```

Reply:
Ah sorry, you are right. My brain was splitting on the filename without extension, which is not the case here.